### PR TITLE
raise when DAGSTER_HOME path is not absolute

### LINKED ISSUE #2609: error on $DAGSTER_HOME set to "."
this doesn't fail in any intuitive way - a lot of things sort of work assuming you run most commands from one directory

### PR DESCRIPTION
**Pull request by AndersonReyes**
fixes #2609

### CODE REVIEW

File: python_modules/dagster/dagster/core/instance/__init__.py [hunk 1]
```diff
@@ -52,7 +52,13 @@ def _dagster_home():
             'DAGSTER_HOME is not set, check is_dagster_home_set before invoking.'
         )
 
-    return os.path.expanduser(dagster_home_path)
+    dagster_home_path = os.path.expanduser(dagster_home_path)
+
+    if not os.path.isabs(dagster_home_path):
+        raise DagsterInvariantViolationError(
+            'DAGSTER_HOME must be absolute path'
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/core/instance/__init__.py:59]: print out what we got here in the error

Comment (AndersonReyes) [reply python_modules/dagster/dagster/core/instance/__init__.py:59]: Gotcha , seems like some tests failed too so I'll double check those too.

Comment (AndersonReyes) [reply python_modules/dagster/dagster/core/instance/__init__.py:59]: I see a lot of the tests use `..` which is causing the test failures can't really find why dagster home gets set that way for other tests

Comment (AndersonReyes) [reply python_modules/dagster/dagster/core/instance/__init__.py]: Actually original ticket was just about `.` but my code tests for both `.` and `..` so I'll remove latter from tests

File: python_modules/dagster/dagster_tests/core_tests/test_instance.py [hunk 2]
```diff
@@ -86,3 +91,13 @@ def noop_pipeline():
 
     assert run.execution_plan_snapshot_id == ep_snapshot_id
     assert run.execution_plan_snapshot_id == create_execution_plan_snapshot_id(ep_snapshot)
+
+
+@pytest.mark.parametrize('dirname', ('.', '..'))
+def test_dagster_home_raises(dirname):
+    os.environ['DAGSTER_HOME'] = dirname
```
Comment (AndersonReyes) [inline python_modules/dagster/dagster_tests/core_tests/test_instance.py:98]: wow, im overriding the os.environ here for the session, use monkeypatch. Had to stare at it for hours before realizing

File: python_modules/dagster/dagster/core/instance/__init__.py [hunk 3]
```diff
@@ -52,7 +52,13 @@ def _dagster_home():
             'DAGSTER_HOME is not set, check is_dagster_home_set before invoking.'
         )
 
-    return os.path.expanduser(dagster_home_path)
+    dagster_home_path = os.path.expanduser(dagster_home_path)
+
+    if not os.path.isabs(dagster_home_path):
+        msg = 'DAGSTER_HOME must be absolute path: {}'.format(dagster_home_path)
+        print(msg)
```
Comment (AndersonReyes) [inline python_modules/dagster/dagster/core/instance/__init__.py:59]: Remove this,

File: python_modules/dagster/dagster_tests/core_tests/test_instance.py [hunk 4]
```diff
@@ -86,3 +91,13 @@ def noop_pipeline():
 
     assert run.execution_plan_snapshot_id == ep_snapshot_id
     assert run.execution_plan_snapshot_id == create_execution_plan_snapshot_id(ep_snapshot)
+
+
+def test_dagster_home_raises(monkeypatch):
+    dirname = '.'
+    monkeypatch.setenv('DAGSTER_HOME', dirname)
```
Comment (alangenfeld) [inline python_modules/dagster/dagster_tests/core_tests/test_instance.py:98]: the approach we use in our tests is a context manager test utility called `environ` - example here https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-celery/dagster_celery_tests/test_config.py#L55

**Comment by alangenfeld** [comment]
Awesome, thanks!

**Pull request merged**